File: solutions/java/lasagna/2/src/main/java/Lasagna.java
public class Lasagna {
    // TODO: define the 'expectedMinutesInOven()' method
    public int expectedMinutesInOven(){
        return 40;
    }

    // TODO: define the 'remainingMinutesInOven()' method
    public int remainingMinutesInOven(int minutePassed){
        return expectedMinutesInOven() - minutePassed;
    }

    // TODO: define the 'preparationTimeInMinutes()' method
    public int preparationTimeInMinutes(int numberOfLayer){
        return numberOfLayer * 2;
    }

    // TODO: define the 'totalTimeInMinutes()' method
    public int totalTimeInMinutes(int numberOfLayer, int minutePassed){
        return minutePassed + preparationTimeInMinutes(numberOfLayer);
    }

}

File: solutions/elixir/lasagna/1/lib/lasagna.ex
defmodule Lasagna do
  # Please define the 'expected_minutes_in_oven/0' function
  def expected_minutes_in_oven() do
    40
  end

  # Please define the 'remaining_minutes_in_oven/1' function
  def remaining_minutes_in_oven(minutes) do 
    40 - minutes
  end
  

  # Please define the 'preparatio9n_time_in_minutes/1' function
  def preparation_time_in_minutes(num_layers) do 
    num_layers * 2
  end

  # Please define the 'total_time_in_minutes/2' function
  def total_time_in_minutes(layers, time_in_oven) do
    layers * 2 + time_in_oven
  end

  # Please define the 'alarm/0' function
  def alarm() do
    "Ding!"
  end
end
